### PR TITLE
🤝 Only show unrevoked shares and unresolved requests on org member dashboard

### PR DESCRIPTION
**Pull request by mermop**
- Only show unrevoked shares and unresolved requests on organisation member dashboard
- Create a feature spec for organisation member dashboard
- Move out the card CSS into its own file

Waiting on a revoked card design from Ross - currently just a red outline wherever it appears

### CODE REVIEW

File: app/models/share.rb [hunk 1]
```diff
@@ -18,10 +18,17 @@ class Share < ApplicationRecord
...
 
+  scope :unrevoked, -> { where(revoked_by: nil) }
+  scope :revoked, -> { where.not(revoked_by: nil) }
+
   def revoked?
     revoked_by.present?
   end
 
+  def unrevoked?
+    !revoked?
```
Comment (Br3nda) [inline app/models/share.rb:29]: :smile_cat:

File: app/views/organisation_member/requests/show.html.erb [hunk 2]
```diff
@@ -44,7 +44,7 @@
         <dd><%= time_ago_in_words(@request.updated_at) %> ago</dd>
       </div>
       <div class="col-sm-6">
-        <% if @request.share.present? %>
+        <% if @request.share.present? && @request.share.unrevoked? %>
```
Comment (Br3nda) [inline app/views/organisation_member/requests/show.html.erb]: how about putting this in Pundit, and instead called `can(:show, @resuest.share)`

Comment (mermop) [reply app/views/organisation_member/requests/show.html.erb]: done in f029340

File: app/policies/share_policy.rb [hunk 3]
```diff
@@ -9,11 +9,11 @@ def initialize(user, share)
   end
 
   def show?
-    user == share.user || user.member_of?(share.recipient)
+    user == share.user || (user.member_of?(share.recipient) && share.unrevoked?)
```
Comment (mermop) [inline app/policies/share_policy.rb:12]: this does bring up a bit of an interesting question actually - when the share is revoked, can the organisation no longer see the share at all, or can they just no longer see the document that's linked to it?

Comment (Br3nda) [reply app/policies/share_policy.rb:12]: I think this is a question for @workbygrant

**Review by Br3nda — APPROVED** [review]

**Pull request merged**